### PR TITLE
 add database tests for albums and faces (#1238)

### LINKED ISSUE #1238: Feat: Add test for database
### Describe the feature

Test coverage is currently 51%, and database functions are not yet covered.

### Add ScreenShots

Image

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

### PR DESCRIPTION
**Pull request by Mansi2007275**
## Summary
Closes #1238

Adds unit tests for the database layer for:
- `backend/app/database/albums.py`
- `backend/app/database/faces.py`

## Test files added
- `backend/tests/test_albums_db.py` — 18 tests
- `backend/tests/test_faces_db.py` — 14 tests

## What is covered
- Table creation functions
- Insert, update, delete operations
- Password hashing and verification
- Edge cases: empty results, missing records, None values
- Batch operations
- Cluster mean embedding calculations

## Notes
- All tests use `unittest.mock` to avoid real DB calls
- Follows existing project test style from `test_albums.py`
- One-line comments above each test describing the scenario

## Summary by CodeRabbit

* **Tests**
  * Added end-to-end SQLite integration tests for album management, covering table creation idempotency, CRUD by id/name, hidden visibility defaults, password hashing/verification, image association ordering, and targeted link removals.
  * Added end-to-end SQLite integration tests for face embeddings, including schema creation idempotency, storage of confidence and JSON bbox, retrieval of unassigned/assigned faces with cluster names, bulk cluster assignment/unassignment behavior, and per-cluster mean embedding computation.

### CODE REVIEW

**Comment by Mansi2007275** [comment]
@rohan-pandeyy  please review my PR.....if its okay then merged it ....

**Comment by rohan-pandeyy** [comment]
Are all the tests that you added passing? Also, it's a kind request to religiously fill the PR template as well. You've removed the entire template which included a lot of crucial point

**Comment by Mansi2007275** [comment]
@coderabbitai review

**Comment by Mansi2007275** [comment]
@rohan-pandeyy CodeRabbit's inline comments keep failing to post (GitHub server error on their end). Could you please take a look directly, or let me know if there's anything specific you'd like changed?

**Comment by Mansi2007275** [comment]
## Type of change
- [x] New tests
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Testing done
- Ran all 37 tests locally with `pytest` — all passing
- Verified linting with `ruff check .` — all checks passed

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix/feature works
- [x] New and existing unit tests pass locally

**Comment by Mansi2007275** [comment]
@rohan-pandeyy I've added the PR template sections (Type of change, Testing done, Checklist) above. All 37 tests are passing and linting is clean. CodeRabbit's comments are still failing to post due to a server error on their end, so please let me know if you spot anything specific that needs fixing, or if you could approve the pending workflow. Thanks for your patience! 🙏

File: backend/tests/test_albums_db.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+class TestDbCreateAlbumsTable:
+    # Tests that albums table is created without raising errors
+    def test_create_albums_table_success(self):
+        with patch("app.database.albums.sqlite3.connect") as mock_connect:
+            mock_conn = MagicMock()
+            mock_connect.return_value = mock_conn
+            db_create_albums_table()
+            mock_conn.cursor.return_value.execute.assert_called_once()
+            mock_conn.commit.assert_called_once()
```
Comment (rohan-pandeyy) [inline backend/tests/test_albums_db.py]: Pytest Fixtures here

File: backend/tests/test_albums_db.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+    # Tests that connection is closed even if an error occurs
+    def test_create_albums_table_closes_connection_on_error(self):
+        with patch("app.database.albums.sqlite3.connect") as mock_connect:
+            mock_conn = MagicMock()
+            mock_conn.cursor.return_value.execute.side_effect = sqlite3.Error("fail")
+            mock_connect.return_value = mock_conn
+            with pytest.raises(sqlite3.Error):
+                db_create_albums_table()
+            mock_conn.close.assert_called_once()
```
Comment (rohan-pandeyy) [inline backend/tests/test_albums_db.py]: Pytest Fixtures here

Comment (rohan-pandeyy) [reply backend/tests/test_albums_db.py]: and so on

**Review by rohan-pandeyy — CHANGES_REQUESTED** [review]
I'd recommend following the same conventions already used in files like `test_albums.py` and `test_face_clusters.py`.

For example, separating fixtures from the actual tests with section headers such as:

```python
# #########################
# Pytest Fixtures
# #########################
```

This helps keep the test files consistent and easier to navigate.

Also, as the test suite for albums and faces continues to grow, it might be worth considering a nested directory structure instead of placing all tests directly under `backend/tests/`. Organizing tests by feature or module would likely make the test suite easier to maintain as it expands.

**Comment by rohan-pandeyy** [comment]
Thanks for your contribution @Mansi2007275 !
